### PR TITLE
tab selection, card cluster selection and description, table sort aria fixes

### PR DESCRIPTION
**Pull request by hitjim**
Previous implementation resulted in screen reader not reading when a tab was selected, or card cluster radio was selected, or table column was sorted.

Changes:
- Use `<button>` instead of `<a>` for tabs
- add `role="tab"`
- detect when `showRadioButton` is used, and apply `role="radio"` or `radiogroup` where applicable. If not, leave as `button`
- create and apply unique `id` for cards in `ts` file instead of repeatedly in the template.
- enhance card cluster to allow using the `name` value (usually used like a description) as `aria-describedby`. This way the shorthand label is still read when describing the radio and its selection, followed by a more detailed description of that selection's purpose.
- tidy up table component's table-related `role`s where possible. Added `aria-sort`-related enums and attempted applying them in the template.

References for proper tab implementation, so as to comply with default screen reader / a11y behavior
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role
- https://developer.mozilla.org/en-US/docs/Learn/Accessibility/HTML
- https://developer.mozilla.org/en-US/docs/Learn/Accessibility/HTML#more_on_links
- https://css-tricks.com/use-button-element/#aa-accessibility-concerns

### CODE REVIEW

**Comment by hitjim** [comment]
edited original PR description for card cluster and table sort commits

**Comment by CharlesSchaack** [comment]
Jim and I had a huddle about the above changes. Everything seems good to me. I give this my approval.

File: src/app/modules/card-cluster/card-cluster.stories.ts
Comment (hitjim) [reply src/app/modules/card-cluster/card-cluster.stories.ts:98]: Chatted a bit - going to put all the `Activate enter to select...` back until/if we have broader investigation and approval for consistent changes in this manner. Commits inbound...

**Review by mcelestini — APPROVED** [review]
👏 dope 👍

**Pull request merged**
